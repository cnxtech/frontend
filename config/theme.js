import theme from '@emcasa/ui'
import {fas} from '@fortawesome/pro-solid-svg-icons'
import {fal} from '@fortawesome/pro-light-svg-icons'
import icons from './icons'

export default {
  ...theme,
  icons: {emcasa: icons, default: fas, light: fal}
}
