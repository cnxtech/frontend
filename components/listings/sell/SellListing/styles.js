import styled from 'styled-components'
import theme from '@emcasa/ui'
import View from '@emcasa/ui-dom/components/View'
import Text from '@emcasa/ui-dom/components/Text'
import {desktopHeaderHeight} from 'constants/dimensions'

const CLOUDINARY_URL = 'https://res.cloudinary.com/emcasa/image/upload/fl_any_format/v1542831121/background'

const UnderlinedText = styled(Text)`
  margin-top: 0;
  text-decoration: underline;
  cursor: pointer;
`

const Container = styled(View)`
  width: 100%;
  display: flex;
  justify-content: center;
  align-self: stretch;
  background-image: url(${CLOUDINARY_URL}/wall-large.jpg);
  background-size: cover;
  background-repeat: no-repeat;
  background-position: bottom;

  @media (max-width: ${theme.breakpoints[0]}){
    background-image: url(${CLOUDINARY_URL}/wall-small.jpg);
  }

  @media (min-width: ${theme.breakpoints[0]}){ 
    @media (max-height: 590px) {
      align-items: center;
      background-image: url(${CLOUDINARY_URL}/wall-small.jpg);
    }
    
    @media (max-width: 1300px) {
      align-items: center;
      margin-top: -20vh;
    }
    
    @media (min-width: 1500px) {
      align-items: center;
      margin-top: -40vh;
    }
  }
  
  @media only screen 
    and (min-device-width : 768px) 
    and (max-device-width : 1024px)  {
    align-items: center;
    background-image: url(${CLOUDINARY_URL}/wall-small.jpg);
  }
`

const Content = styled(View)`
  max-width: 100%;
  margin-top: ${desktopHeaderHeight * 2}px;
   @media (min-width: ${theme.breakpoints[0]}){ 
    @media (min-aspect-ratio: 2 / 1) {
      margin-top: ${desktopHeaderHeight}px;    
    }
   }
`

export {
  UnderlinedText,
  Container,
  Content
}
