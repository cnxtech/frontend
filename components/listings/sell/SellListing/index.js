import {Component} from 'react'
import Link from 'next/link'
import Router from 'next/router'
import Col from '@emcasa/ui-dom/components/Col'
import Row from '@emcasa/ui-dom/components/Row'
import Text from '@emcasa/ui-dom/components/Text'
import Button from '@emcasa/ui-dom/components/Button'
import {isMobile} from 'lib/mobile'
import AddressAutoComplete from 'components/shared/AddressAutoComplete'
import MobileAddressButton from 'components/shared/MobileAddressButton'
import { MobieTypeaheadContainer } from 'components/shared/AddressAutoComplete/styles'

import {
  UnderlinedText,
  Container,
  Content
} from './styles'

export default class SellListing extends Component {
  constructor(props) {
    super(props)
    this.openMobileAddressInput = this.openMobileAddressInput.bind(this)
    this.close = this.close.bind(this)
  }

  state = {
    showMobileAddressInput: false,
    addressFormatted: null,
    addressData: null
  }

  openMobileAddressInput() {
    this.setState({
      showMobileAddressInput: true
    })
  }

  close() {
    this.setState({
      showMobileAddressInput: false
    })
  }

  render() {
    if (this.state.showMobileAddressInput) {
      return (
        <MobieTypeaheadContainer justifyContent="center" p={4}>
          <Col width={1}>
            <AddressAutoComplete
              onBackPressed={this.close}
              defaultValue={this.state.addressFormatted}
              onClearInput={() => {}}
              onSelectAddress={(addressFormatted, addressData) => {
                this.setState({
                  showMobileAddressInput: false,
                  addressFormatted,
                  addressData
                })
              }}
            />
          </Col>
        </MobieTypeaheadContainer>
      )
    }

    return (
      <Container>
        <Content>
          <Row justifyContent="center" px={[4, 0]}>
            <Col>
              <Text fontSize="xlarge" color="dark" fontWeight="bold" textAlign="center">
                Quer vender seu imóvel?
              </Text>
            </Col>
          </Row>
          <Row justifyContent="center" px={[4, 0]}>
            <Col>
              <Text color="gray">
                Vender seu imóvel nunca foi tão fácil, informe o endereço do seu imóvel
                e receba agora uma avaliação grátis
              </Text>
            </Col>
          </Row>
          <Row flexDirection={['column', 'row']} px={[4, 0]}>
            <Col width={[1, 10 / 12]} mr={[0, 2]}>
              {isMobile() ?
                <Col mb={4}>
                  <MobileAddressButton
                    onClick={this.openMobileAddressInput}
                    address={this.state.addressFormatted}
                  />
                </Col>
              :
                <Col mb={4} width={1}>
                  <AddressAutoComplete
                    defaultValue={this.state.address}
                    onClearInput={this.onClearInput}
                    onSelectAddress={(addressFormatted, addressData) => {
                      this.setState({
                        addressFormatted,
                        addressData
                      })
                    }}
                  />
                </Col>
              }
            </Col>
            <Col width={[1, 2 / 12]}>
              <Button height="tall" active fluid onClick={() => {
                localStorage.setItem('sellerAddressFormatted', this.state.addressFormatted)
                localStorage.setItem('sellerAddressData', JSON.stringify(this.state.addressData))
                Router.push('/vender/imovel')
              }}>
                Avaliar
              </Button>
            </Col>
          </Row>
          <Row justifyContent="center">
            <Col>
              <Link href="/">
                <UnderlinedText color="pink">Você quer comprar?</UnderlinedText>
              </Link>
            </Col>
          </Row>
        </Content>
      </Container>
    )
  }
}
