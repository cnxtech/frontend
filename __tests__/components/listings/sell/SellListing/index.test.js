import React from 'react'
import SellListing from 'components/listings/sell/SellListing'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import theme from 'config/theme'
import { ThemeProvider } from 'styled-components'

describe('<SellListing/>', () => {
  it('should render a clean page', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <SellListing />
      </ThemeProvider>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
