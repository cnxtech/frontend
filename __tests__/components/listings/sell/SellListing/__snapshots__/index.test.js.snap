// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SellListing/> should render a clean page 1`] = `
.c6 {
  margin-bottom: 20px;
  margin-right: 10px;
  width: 100%;
}

.c7 {
  margin-bottom: 20px;
}

.c10 {
  width: 100%;
}

.c13 {
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  -webkit-font-smoothing: antialiased;
  font-family: FaktSoftPro-Normal;
  color: #3c4858;
  font-size: 32px;
  font-weight: bold;
  text-align: center;
}

.c4 {
  -webkit-font-smoothing: antialiased;
  font-family: FaktSoftPro-Normal;
  color: gray;
  font-size: 18px;
}

.c16 {
  -webkit-font-smoothing: antialiased;
  font-family: FaktSoftPro-Normal;
  color: #f50057;
  font-size: 18px;
}

.c14 {
  outline: none;
  display: block;
  cursor: pointer;
  padding: 0 20px 0 20px;
  border-radius: 4px;
  border: 1px solid;
  border-color: #f50057;
  background-color: #f50057;
  width: 100%;
  height: 60px;
  color: #ffffff;
  font-family: FaktSoftPro-Normal;
  font-size: 18px;
  -webkit-font-smoothing: antialiased;
  border-color: #e0e6ed;
}

.c14 :hover {
  background-color: #f50057;
}

.c12 {
  margin-right: 10px;
}

.c11 {
  -webkit-font-smoothing: antialiased;
  display: block;
  box-sizing: border-box;
  width: 100%;
  outline: none;
  border-radius: 4px;
  border: 1px solid;
  border-color: #e0e6ed;
  padding: 0px 10px 0px 10px;
  height: 60px;
  color: #3c4858;
  font-family: FaktSoftPro-Normal;
  font-size: 18px;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
}

.c11:focus {
  border-radius: 4px;
  border: 1px solid;
  border-color: #1fb6ff;
  padding: 0px 10px 0px 10px;
  height: 60px;
}

.c11::-webkit-input-placeholder {
  color: #8492a6;
}

.c11::-moz-placeholder {
  color: #8492a6;
}

.c11:-ms-input-placeholder {
  color: #8492a6;
}

.c11::placeholder {
  color: #8492a6;
}

.c8 {
  border: 1px solid #e0e6ed;
  border-radius: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c15 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  background-image: url(https://res.cloudinary.com/emcasa/image/upload/fl_any_format/v1542831121/background/wall-large.jpg);
  background-size: cover;
  background-repeat: no-repeat;
  background-position: bottom;
}

.c1 {
  margin-top: 76px;
  max-width: 700px;
}

@media screen and (min-width:640px) {
  .c6 {
    margin-bottom: 0px;
  }
}

@media screen and (min-width:640px) {
  .c6 {
    width: 83.33333333333334%;
  }
}

@media screen and (min-width:640px) {
  .c13 {
    width: 16.666666666666664%;
  }
}

@media screen and (min-width:640px) {
  .c5 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (max-width:640px) {
  .c0 {
    background-image: url(https://res.cloudinary.com/emcasa/image/upload/fl_any_format/v1542831121/background/wall-small.jpg);
  }
}

@media (min-aspect-ratio:2 / 1) {
  .c0 {
    background-image: url(https://res.cloudinary.com/emcasa/image/upload/fl_any_format/v1542831121/background/wall-small.jpg);
  }
}

@media (max-width:640px) {
  .c1 {
    padding-right: 20px;
    padding-left: 20px;
  }
}

<div
  className="c0 "
>
  <div
    className="c1 "
  >
    <div
      className="c2"
    >
      <div
        className=""
      >
        <p
          className="c3"
          color="dark"
          fontSize="xlarge"
          fontWeight="bold"
        >
          Quer vender seu imóvel?
        </p>
      </div>
    </div>
    <div
      className="c2"
    >
      <div
        className=""
      >
        <p
          className="c4"
          color="gray"
          fontSize="default"
        >
          Vender seu imóvel nunca foi tão fácil, informe o endereço do seu imóvel e receba agora uma avaliação grátis
        </p>
      </div>
    </div>
    <div
      className="c5"
    >
      <div
        className="c6"
        width={
          Array [
            1,
            0.8333333333333334,
          ]
        }
      >
        <div
          className="c7"
        >
          <div
            className="c8 c9"
          >
            <div
              className="c10"
              width={1}
            >
              <div
                className=""
              >
                <input
                  autoComplete="off"
                  className="c11"
                  fontSize="default"
                  height="tall"
                  letterSpacing="default"
                  name="street"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Endereço e número*"
                  style={
                    Object {
                      "border": 0,
                    }
                  }
                  type="text"
                  value={null}
                />
              </div>
            </div>
            <div
              className="c12"
            />
          </div>
        </div>
      </div>
      <div
        className="c13"
        width={
          Array [
            1,
            0.16666666666666666,
          ]
        }
      >
        <button
          className="c14"
          fontSize="default"
          height="tall"
        >
          Avaliar
        </button>
      </div>
    </div>
    <div
      className="c2"
    >
      <div
        className=""
      >
        <p
          className="c15 c16"
          color="pink"
          fontSize="default"
          onClick={[Function]}
        >
          Você quer comprar?
        </p>
      </div>
    </div>
  </div>
</div>
`;
